src/Sentry/TransactionTracer.cs

```diff
@@ -210,59 +207,75 @@ public TransactionTracer(IHub hub, ITransactionContext context)
         Description = context.Description;
         Status = context.Status;
         IsSampled = context.IsSampled;
+        StartTimestamp = _stopwatch.StartDateTimeOffset;
+
+        if (context is TransactionContext transactionContext)
+        {
+            _instrumenter = transactionContext.Instrumenter;
+        }
     }
 
     /// <inheritdoc />
-    public void AddBreadcrumb(Breadcrumb breadcrumb) =>
-        _breadcrumbs.Add(breadcrumb);
+    public void AddBreadcrumb(Breadcrumb breadcrumb) => _breadcrumbs.Add(breadcrumb);
 
     /// <inheritdoc />
-    public void SetExtra(string key, object? value) =>
-        _extra[key] = value;
+    public void SetExtra(string key, object? value) => _extra[key] = value;
 
     /// <inheritdoc />
-    public void SetTag(string key, string value) =>
-        _tags[key] = value;
+    public void SetTag(string key, string value) => _tags[key] = value;
 
     /// <inheritdoc />
-    public void UnsetTag(string key) =>
-        _tags.TryRemove(key, out _);
+    public void UnsetTag(string key) => _tags.TryRemove(key, out _);
 
     /// <inheritdoc />
     [EditorBrowsable(EditorBrowsableState.Never)]
-    public void SetMeasurement(string name, Measurement measurement) =>
-        _measurements[name] = measurement;
+    public void SetMeasurement(string name, Measurement measurement) => _measurements[name] = measurement;
 
-    internal ISpan StartChild(SpanId parentSpanId, string operation)
-    {
-        // Limit spans to 1000
-        var isOutOfLimit = _spans.Count >= 1000;
+    /// <inheritdoc />
+    public ISpan StartChild(string operation) => StartChild(SpanId, operation);
 
-        var span = new SpanTracer(_hub, this, parentSpanId, TraceId, operation)
+    internal ISpan StartChild(SpanId parentSpanId, string operation, Instrumenter instrumenter = Instrumenter.Sentry) =>
```

Comment:
Do we need an overload here? If we can avoid having internal overloads that'd be better (keeping the API even if internal, simpler. Easier to change/reason about, etc).

I think we can keep the one below only and be explicit with our `null` on the `spanId` is we need to.